src/data_provider/src/packages/appxWindowsWrapper.h

```diff
@@ -0,0 +1,401 @@
+            {
+                architecture = "i686";
+            }
+
+            return architecture;
+        }
+
+        /*
+         * Getting location where is the data and configuration application.
+         *
+         * @param[in] registry  Registry object pointing to where the data is.
+         *
+         * @return Return package location.
+         */
+        const std::string getLocation(const Utils::Registry& registry)
+        {
+            std::string location;
+
+            return registry.string("PackageRootFolder", location) ? location : UNKNOWN_VALUE;
+        }
+
+        /*
+         * @brief Get application name.
+         *
+         * @param[in] fullName Full application name.
+         * @param[in] registry Registry object pointing to where the data is located.
+         *
+         * @return Return name application.
+         */
+        const std::string getName(const std::string& fullName, const Utils::Registry& registry)
+        {
+            std::string name;
+
+            /*
+             * Example of  fullName
+             *
+             * Microsoft.Windows.Search
+             * Microsoft.SkypeApp
+             */
+            const auto keyName { Utils::split(fullName, '.').back() }; // Will only use the last vector element.
+            constexpr auto INVALID_NAME_APP { "@{" };
+
+            for (const auto& folder : registry.enumerate())
+            {
+                try
+                {
+                    std::string value;
+                    const Utils::Registry nameReg(m_key, m_userId  + "\\" + APPLICATION_STORE_REGISTRY + "\\" + m_nameApp + "\\" + folder + "\\Capabilities");
+
+                    if (nameReg.string("ApplicationName", value))
+                    {
+                        name = value;
+                    }
+                }
+                catch (...)
+                {
+                }
+            }
+
+            const auto pos { name.find(INVALID_NAME_APP) };
+
+            /*
+             * If the name starts with "@{" string then we need to look for the name on the cache registry.
+             * The name obtained is the key for the app name.
+             */
+            if (pos != std::string::npos && pos == 0)
+            {
+                try
+                {
+                    name = searchNameFromCacheRegistry(keyName, name);
+                }
+                catch (...)
+                {
+                }
+            }
+
+            return name;
+        }
+
+        /*
+         * @brief Get application vendor name.
+         *
+         * @param[in] registry Registry object that points to where the sub-registries with the names of the registry with the data are located.
+         *
+         * @return Returns the vendor name
+         */
+        const std::string getVendor(const Utils::Registry& registry)
+        {
+            std::string vendor;
+
+            for (const auto& folder : registry.enumerate())
```

Comment:
registry() can throw an exception let's catch it please and return the default value.